### PR TITLE
feat(template): add new shiny template app

### PR DESCRIPTION
**Pull request by barsukov**
# Summary

This pull request introduces a new template app that incorporates several modern tools and technologies to streamline the development process. It serves as a foundation for our future projects, including our carbon app.

# Changes Made

- Added a new template application that uses Vite, ReactJS, and TypeScript.
- Integrated Vitest and Testing Library for testing.
- Configured TailwindCSS for styling and implemented Juno UI components.
- Set up Eslint and Prettier for consistent code quality and formatting.
- Added `turbo` commands to simplify the project setup and generation.
- Husky pre-commit lint fix remove 

# Related Issues

- # https://github.com/cloudoperators/juno/issues/452

# Screenshots (if applicable)

image

# Testing Instructions

1. `npm run generate:app juno-new-app`
2. `cd apps/juno-new-app turbo dev`

# Checklist

- [x] I have performed a self-review of my code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have added tests that prove my fix is effective or that my feature works.
- [x] New and existing unit tests pass locally with my changes.
- [x] I have made corresponding changes to the documentation (if applicable).
- [x] My changes generate no new warnings or errors.

# Tools Used

This project uses many tools, including:

- [Vite](https://vitejs.dev)
- [ReactJS](https://reactjs.org)
- [TypeScript](https://www.typescriptlang.org)
- [Vitest](https://vitest.dev)
- [Testing Library](https://testing-library.com)
- [TailwindCSS](https://tailwindcss.com)
- [Eslint](https://eslint.org)
- [Prettier](https://prettier.io)
- [Juno UI components](https://github.com/cloudoperators/juno/tree/main/packages/juno-ui-components)

This setup allows us to easily generate everything with a single root command task using `turbo`. It's also a great starting point for building our carbon app.

### CODE REVIEW

File: apps/template/.editorconfig [hunk 1]
```diff
@@ -0,0 +1,9 @@
+root = true
```
Comment (andypf) [inline apps/template/.editorconfig:1]: Are this settings for VSCode?

File: apps/template/package.json [hunk 2]
```diff
@@ -15,70 +11,48 @@
...
-      "value": "",
-      "type": "required",
-      "description": "Endpoint URL of the API"
-    },
-    "embedded": {
-      "value": "false",
-      "type": "optional",
-      "description": "Set to true if app is to be embedded in another existing app or page.  If set to true the app won't render a page header/footer and instead render only the content. The default value is false."
-    }
+    "prop-types": "15.8.1",
```
Comment (andypf) [inline apps/template/package.json:26]: Let's don't use deprecated packages like prop-types, even if we use typescript

**Comment by andypf** [comment]
we could move favicon to src/images, then we don't need public folder at all

**Comment by andypf** [comment]
and good job 👍

**Review by andypf — APPROVED** [review]

**Review by andypf — DISMISSED** [review]

File: apps/template/.gitignore [hunk 3]
```diff
@@ -1,3 +1,37 @@
-node_modules
-secretProps
```
Comment (ArtieReus) [inline apps/template/.gitignore:2]: I would add default appProps instead

File: apps/template/index.html [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+<html lang="en">
+  <head>
+    <meta charset="UTF-8" />
+    <link rel="icon" type="image/svg+xml" href="src/assets/favicon.ico" />
+    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
+    <title>React 18, TypeScript, Vitest, Testing Library, TailwindCSS 3, Vite, Eslint, Juno UI components and Prettier boilerplate</title>
+  </head>
+  <body>
+    <div id="root"></div>
+    <script type="module" src="/src/index.tsx"></script>
```
Comment (ArtieReus) [inline apps/template/index.html:11]: I'd like to demonstrate how we pass appProps to the application, ensuring that this template is pre-configured when used for any future application. Please see the example App or doop:
https://github.com/cloudoperators/juno/blob/main/apps/example/index.html#L34

File: apps/template/package.json [hunk 5]
```diff
@@ -15,70 +11,48 @@
...
-    "util": "^0.12.4"
-  },
   "scripts": {
-    "build": "NODE_ENV=production node esbuild.config.js",
-    "dev": "NODE_ENV=development node esbuild.config.js --serve --watch",
-    "lint": "eslint .",
-    "test": "jest"
+    "dev": "vite",
+    "build": "vite build",
+    "serve": "vite preview",
```
Comment (ArtieReus) [inline apps/template/package.json:17]: I called this task preview on doop. We might have to agree how to call this task.

File: apps/template/package.json [hunk 6]
```diff
@@ -15,70 +11,48 @@
...
-    "tailwindcss": "^3.3.1",
-    "util": "^0.12.4"
-  },
   "scripts": {
-    "build": "NODE_ENV=production node esbuild.config.js",
-    "dev": "NODE_ENV=development node esbuild.config.js --serve --watch",
-    "lint": "eslint .",
-    "test": "jest"
+    "dev": "vite",
+    "build": "vite build",
```
Comment (ArtieReus) [inline apps/template/package.json:16]: What about adding also the task build:static?

File: apps/template/secretProps.template.json
Comment (ArtieReus) [inline apps/template/secretProps.template.json:1]: What about having a appProps.template.json

**Review by ArtieReus — CHANGES_REQUESTED** [review]
Overall, it looks good. I think we should agree on a default setup for our apps to ensure consistency across all of them. Currently, the template app has a slightly different setup compared to Doop or the example app.

File: apps/template/index.html (hunk 4 above)
Comment (barsukov) [reply apps/template/index.html:11]: Why do we need that in a first place?

File: apps/template/package.json (hunk 5 above)
Comment (barsukov) [reply apps/template/package.json:17]: Yes, lets agree

File: apps/template/package.json (hunk 6 above)
Comment (barsukov) [reply apps/template/package.json:16]: The usecase is for preview app right? I would like to investigate here more on this one, but I could add one no problem.

File: apps/template/secretProps.template.json
Comment (barsukov) [reply apps/template/secretProps.template.json:1]: why do we need it?

**Review by andypf — APPROVED** [review]

**Review by ArtieReus — APPROVED** [review]
lgtm

**Pull request merged**